lib/modules/manager/velaci/extract.ts

```diff
@@ -0,0 +1,60 @@
+import { load } from 'js-yaml';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type { PackageDependency, PackageFile } from '../types';
+import type { VelaPipelineConfiguration } from './types';
+
+export function extractPackageFile(file: string): PackageFile | null {
+  let doc: VelaPipelineConfiguration;
+
+  try {
+    doc = load(file, { json: true }) as VelaPipelineConfiguration;
+  } catch (err) {
+    logger.warn({ err, file }, 'Failed to parse Vela file.');
+    return null;
+  }
+
+  const deps: PackageDependency[] = [];
+  try {
+    // iterate over steps
+    doc.steps?.forEach((step) => {
+      const dep: PackageDependency = getDep(step.image);
+
+      deps.push(dep);
+    });
+
+    // iterate over services
+    doc.services?.forEach((service) => {
+      const dep: PackageDependency = getDep(service.image);
+
+      deps.push(dep);
+    });
+
+    // iterate over stages
+    for (const stage in doc.stages) {
```

Comment:
```suggestion
    for (const [name, stage] of Object.entries(doc.stages)) {
```

Reply:
done, though i skipped `name` since we don't need to use it